settings.yml

```diff
@@ -57,15 +57,15 @@
   
   # Either `true` to allow squash-merging pull requests, or `false` to prevent
   # squash-merging.
-  #allow_squash_merge: true
+  allow_squash_merge: false
   
   # Either `true` to allow merging pull requests with a merge commit, or `false`
   # to prevent merging pull requests with merge commits.
-  #allow_merge_commit: true
+  allow_merge_commit: true
   
   # Either `true` to allow rebase-merging pull requests, or `false` to prevent
   # rebase-merging.
-  #allow_rebase_merge: true
+  allow_rebase_merge: false
```

Comment:
why not? I is quite useful to being able to rebase using UI.

Reply:
Well I initially wanted to do this so there is a clear history on what's happened to the code and also what PR it came from. I was concerned there were some security implications on changing the repo's history. For example if in some documentation it linked to a commit which no longer exists. That may be confusing/annoying. That's not a security issue and now that I think about it I can't really think of security issues. But I may have the idea of rebasing in my head wrong.

Reply:
> But I may have the idea of rebasing in my head wrong.

Indeed, but github docs are misleading a bit about that (https://github.com/github/docs/issues/17959). It won't change existing history on our repository in the case someone has a PR from his fork - it will change history of the feature/fix branch on HIS fork (in general: head branch), so this definitely not an intrusive way interacting with our base branch.

This is more less equivalent to workflow like that:
```bash
git checkout -b new_stuff
(hack hack & commit)
git fetch origin main
git rebase origin/main
git push -f   # rewrite history on new_stuff branch before we can cleanly add all commits on top of main
```

Reply:
This article may help you to decide what you prefer. I have used the merge approach more often, but either can work. The larger the group of developers working in parallel the more often I have used the merge approach.
https://www.atlassian.com/git/articles/git-team-workflows-merge-or-rebase

Reply:
Yeah I think this can be discussed more later but for the purpose of the safe-settings config, I think allowing all 3 seems about right.